### PR TITLE
Weap-Structure, New engine client, replaced Unrestricted.

### PR DESCRIPTION
**Pull request by cantbeatme**
Fixes triggerbot & autowall issues

### CODE REVIEW

**Comment by CraftLourens** [comment]
Wow, thank you. You're spoonfeeding me but I can't meik it work by myself :disappointed:

**Comment by cantbeatme** [comment]
Np. Ignore updates for endscene & netvar managers.
They were edited because I've done some pretty large edits and added a few new features to the cheat, didn't mean to push them.

**Comment by CraftLourens** [comment]
Honestly I removed parts that you updates ages ago since I don't need them.

File: csgo_internal/csgo_internal/IEngineClient.h [hunk 1]
```diff
@@ -137,7 +137,11 @@ class IEngineClient
 	virtual int                   GetAppID() = 0;
 	virtual Vector                GetLightForPointFast( const Vector &pos, bool bClamp ) = 0;
 	virtual void                  ClientCmd_Unrestricted1( char  const*, int, bool );
-	virtual void                  ClientCmd_Unrestricted( const char *szCmdString ) = 0; // 114
+	virtual void                  ClientCmd_Unrestricted( const char *szCmdString, const char *newFlag) = 0; // 114, new flag, quick testing shows setting 0 seems to work, haven't looked into it.
+	//Forgot to add this line, but make sure to format all unrestricted calls now with an extra , 0
+	//Ex:
+	//	I::Engine->ClientCmd_Unrestricted( charenc( "cl_mouseenable 1" ) , 0);
+	//	I::Engine->ClientCmd_Unrestricted( charenc( "crosshair 1" ) , 0);
```
Comment (QWp6t) [inline csgo_internal/csgo_internal/IEngineClient.h]: You could also leave the definition unmodified, and instead add this to IEngineClient.cpp

```c++
void IEngineClient::ClientCmd_Unrestricted(const char *szCmdString)
{
	typedef void(__thiscall* OriginalFn)(PVOID, const char*, char);
	return U::GetVFunc<OriginalFn>(this, 114)(this, szCmdString, 0);
}
```

Poached from UC: https://www.unknowncheats.me/forum/1515371-post5207.html

File: csgo_internal/csgo_internal/Entity.cpp [hunk 2]
```diff
@@ -380,6 +380,7 @@ bool CBaseCombatWeapon::IsGun()
 		return false;
 
 	int id = this->GetWeaponID();
+//If your aimbot is broken, this is the reason. Just an FYI.
```
Comment (QWp6t) [inline csgo_internal/csgo_internal/Entity.cpp]:
```c++
	offsets.m_iWeaponID = U::NetVars->GetOffset( charenc( "DT_WeaponCSBase" ), charenc( "m_fAccuracyPenalty" ) ) + 0x30;
```

Comment (cantbeatme) [inline csgo_internal/csgo_internal/Entity.cpp]: I switched to ItemDefinitionIndex, provides more options.

**Pull request merged**